Android/app/src/main/java/com/example/androidgui/deliverystaffactivities/DeliverOrderActivity.java

```diff
@@ -1,24 +1,35 @@
 package com.example.androidgui.deliverystaffactivities;
 
+import android.annotation.SuppressLint;
 import android.content.Intent;
+import android.net.Uri;
 import android.view.View;
 import android.widget.TextView;
 import android.widget.Toast;
 import androidx.appcompat.app.AppCompatActivity;
 import android.os.Bundle;
 import com.example.androidgui.R;
-import presenter.staffsystem.StaffController;
+import presenter.staffsystem.GeoDestination;
+import presenter.staffsystem.StaffPresenter;
+import presenter.staffsystem.StaffViewInterface;
 
-public class DeliverOrderActivity extends AppCompatActivity {
+import java.util.Objects;
+
+public class DeliverOrderActivity extends AppCompatActivity implements StaffViewInterface, GeoDestination {
     private String id;
     private String mode;
-    private StaffController controller;
+    private StaffPresenter controller;
+    private TextView currentOrder;
+    private String destination;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
-        controller = new StaffController();
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_deliver_order);
+        // Setup view bonded attributes
+        controller = new StaffPresenter();
```

Comment:
might want to change this name to presenter otherwise it is kind of confusing

Reply:
Finished, see commit 238e5c11d72a64e52c0d4e6d186172aef4f21591.